tests/phpunit/tests/blocks/wpBlockType.php

```diff
@@ -460,4 +460,47 @@ public function data_block_version() {
 			array( '<!- - wp:core/separator -->', 0 ),
 		);
 	}
+
+	/**
+	 * @ticket 59969
+	 */
+	public function test_variation_callback() {
+		$block_type = new WP_Block_Type(
+			'test/block',
+			array(
+				'title'              => 'Test title',
+				'variation_callback' => array( $this, 'mock_variation_callback' ),
+			)
+		);
+
+		$this->assertSameSets( $this->mock_variation_callback(), $block_type->variations );
+	}
+
+	/**
+	 * @ticket 59969
+	 * @covers WP_Block_Type::get_variations
+	 */
+	public function test_get_variations() {
+		$block_type = new WP_Block_Type(
+			'test/block',
+			array(
+				'title'              => 'Test title',
+				'variation_callback' => array( $this, 'mock_variation_callback' ),
+			)
+		);
+
+		$this->assertSameSets( $this->mock_variation_callback(), $block_type->get_variations() );
+	}
```

Comment:
We should probably add a few additional unit tests here to account for cases where the variations are already set, and when something is hooked into the variations callback to demonstrate the expected behavior that the callback will only be run on the first iteration and that manually set variations will override the existence of a variations callback if not already called, but those can be added in an additional commit if needed.

Reply:
I'll add it first thing in the morning tomorrow so we can have it in the final merge.

Reply:
@joemcgill I have added 7 more test cases to cover such cases.